analyzers/tests/SonarAnalyzer.UnitTest/MetadataReferences/NuGetMetadataFactory.Package.cs

```diff
@@ -20,152 +20,132 @@
                 }
+                WriteNextCheckTime();
+            }
 
-                static string ValidatedNuGetConfigPath()
-                {
-                    var path = Path.GetFullPath(NuGetConfigFileRelativePath);
-                    if (!File.Exists(path))
-                    {
-                        throw new ApplicationException($"Test setup error: failed to find nuget.config file at \"{path}\"");
-                    }
-                    LogMessage($"Path to nuget.config: {path}");
-                    return path;
-                }
+            private static async Task<FindPackageByIdResource> GetNuGetRepository()
+            {
+                var nugetOrgUrl = Settings.LoadSpecificSettings(NugetConfigFolderRelativePath, "nuget.config")
+                                          .GetSection("packageSources").Items.OfType<AddItem>()
+                                          .Where(ai => ai.Key == "nuget.org")
+                                          .Select(ai => ai.Value)
+                                          .Single();
+                var repository = Repository.Factory.GetCoreV3(nugetOrgUrl);
+                return await repository.GetResourceAsync<FindPackageByIdResource>();
             }
 
             /// <summary>
             /// Returns the list of folders containing installed versions of the specified package,
             /// or an empty list if the package is not installed.
             /// </summary>
             /// <remarks>
-            /// Package directory names are in the form "{package id}.{package version}".
+            /// Package directory names are in the form "{package id}\{package version}".
             /// The list is sorted in ascending order, so the most recent version will be last.
             /// </remarks>
             private IEnumerable<string> SortedPackageFolders()
             {
-                // The package will be in a folder called "\packages\{packageId}.{version}", but:
+                // The package will be in a folder called "\packages\{packageId}\{version}", but:
                 // : the package might not be installed
                 // : there might be multiple versions installed
                 // : there might be a package that starts with the same package id
                 //      e.g. Microsoft.AspNetCore.Core and Microsoft.AspNetCore.Core.Diagnostics
                 // Most packages have a three-part version, but some have four. We don't check
                 // the actual number of parts, as long as there is at least one.
-                var matcher = new Regex($@"{Regex.Escape(Id)}(\.\d+)+$", RegexOptions.IgnoreCase);
-
-                return Directory.Exists(PackagesFolder)
-                    ? Directory.GetDirectories(PackagesFolder, $"{Id}.*", SearchOption.TopDirectoryOnly).Where(x => matcher.IsMatch(x)).OrderBy(x => x)
+                var matcher = new Regex(@"(\.\d+)+$", RegexOptions.IgnoreCase);
```

Comment:
Lines 119 and 120 above are now out of date and can be removed. It's not a concern anymore with the subdirectory.